app/controllers/users/invitations_controller.rb

```diff
@@ -26,6 +26,7 @@ def validate_invited_user
   end
 
   def user_is_invalid?
+    delete_user_record if invited_user_already_exists? && invited_user_not_confirmed?
```

Comment:
it isn't really the responsibility of the validation check to delete the user record.

It might be worth moving this to the same level as the `resending_invite?` is at:

```rb
before_action :delete_user_record, if: :user_should_be_cleared?, only: :create

def user_should_be_cleared?
  resending_invite? or invited_user_not_confirmed?
end
```